extension/zpagesextension/zpagesextension.go

```diff
@@ -73,7 +73,12 @@ func (zpe *zpagesExtension) Start(_ context.Context, host component.Host) error
 	}
 
 	zpe.telemetry.Logger.Info("Starting zPages extension", zap.Any("config", zpe.config))
-	zpe.server = http.Server{Handler: zPagesMux}
+	server, err := zpe.config.ToServer(host, zpe.telemetry, nil)
```

Comment:
Do you not want to pass the mux that serves http?
```suggestion
	server, err := zpe.config.ToServer(host, zpe.telemetry, zPagesMux)
```

Reply:
Actually I tried this but again the same test is failing, I think I will have to look at the implementation carefully or may be there are some more things I would need to explore in order to address  the Z-page extension server using configHttp module